### PR TITLE
Truncate-tooltip optimization

### PR DESCRIPTION
**Pull request by galloween**
some optimizations towards more css-based solution

### CODE REVIEW

File: projects/ui-framework/src/lib/services/truncate-tooltip/truncate-tooltip.component.ts [hunk 1]
```diff
@@ -63,18 +52,25 @@ export class TruncateTooltipComponent
     this.setMaxLines(value);
   }
   @Input() delay = 300;
+  @Input() expectChanges = true;
+  @Input() useCssVars = false;
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/services/truncate-tooltip/truncate-tooltip.component.ts]: I think the default values should be the more optimised option, meaning the user should input if he expectsChanges or not, and if using the CssVars is better performance, maybe this should be the default. wdyt?

File: projects/ui-framework/src/lib/style/functions.scss [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        @if $get == null {
+          @return null;
+        }
+      } @else {
+        @return null;
+      }
+    }
+  }
+  @return $get;
+}
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/style/functions.scss]: what does this return?

Comment (galloween) [reply projects/ui-framework/src/lib/style/functions.scss]: this function returns deep-nested key
used like  map-get-deep($props, 'text' 'text-slot-1' 'font')
= lodash  get($props, '.text.textSlot1.font', null)

its not used in tooltip, but in avarar.

File: projects/ui-framework/src/lib/services/truncate-tooltip/truncate-tooltip.component.ts (hunk 1 above)
Comment (galloween) [reply projects/ui-framework/src/lib/services/truncate-tooltip/truncate-tooltip.component.ts]: then it means we need to enable the typography mixins to insert variables everywhere.
if somebody is not using mixins or setting font size manually, like font-size: 12px, it wont work...

**Pull request merged**